packages/main/src/backend/import/bankScraper.ts

```diff
@@ -23,20 +22,38 @@ export async function scrape(
   { companyId, credentials, startDate, timeout, showBrowser = false }: ScrapeParameters,
   emitProgressEvent: EmitProgressEventFunction,
   chromePath: string,
+  eventPublisher: EventPublisher,
 ) {
   const options: ScraperOptions = {
-    companyId, // mandatory; one of 'hapoalim', 'discount', 'otsarHahayal', 'leumiCard', 'isracard', 'amex'
-    startDate, // the date to fetch transactions from (can't be before the minimum allowed time difference for the scraper)
-    combineInstallments: false, // if set to true, all installment transactions will be combine into the first one
-    showBrowser, // shows the browser while scraping, good for debugging (default false)
-    verbose: false, // include more debug info about in the output
+    companyId,
+    startDate,
+    combineInstallments: false,
+    showBrowser,
+    verbose: false,
     executablePath: chromePath,
     defaultTimeout: timeout,
   };
+
   const scraper = createScraper(options);
-  scraper.onProgress((eventCompanyId: string, payload: { type: string }) => {
-    emitProgressEvent(companyId, payload.type);
-  });
-  const scrapeResult = await scraper.scrape(credentials);
-  return scrapeResult;
+  scraper.onProgress((eventCompanyId, payload) => emitProgressEvent(companyId, payload.type));
+
+  if (companyId === CompanyTypes.oneZero) {
+    const creds = credentials as typeof credentials & { otpLongTermToken: string; phoneNumber: string };
+    if (!creds.otpLongTermToken) {
+      await scraper.triggerTwoFactorAuth(creds.phoneNumber);
+      await eventPublisher.emit(EventNames.GET_OTP);
```

Comment:
1. Use `emitProgressEvent` instead of `eventPublisher.emit`.
1. Is the `scraper` not emitting events of OTP itself?